From 84b0551eb141b3a7904ceec158d040200c995023 Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Thu, 8 Apr 2021 22:07:15 +0200
Subject: [PATCH] Fix crash with newer Qt versions

Assign m_internalModel before calling into
QSortFilterProxyModel::setSourceModel since that can end up calling filterAcceptsRow
---
 src/gui/file/sortfilterfilemodel.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/file/sortfilterfilemodel.cpp b/src/gui/file/sortfilterfilemodel.cpp
index 73707cec..87657aa2 100644
--- a/src/gui/file/sortfilterfilemodel.cpp
+++ b/src/gui/file/sortfilterfilemodel.cpp
@@ -40,8 +40,8 @@ SortFilterFileModel::SortFilterFileModel(QObject *parent)
 
 void SortFilterFileModel::setSourceModel(QAbstractItemModel *model)
 {
-    QSortFilterProxyModel::setSourceModel(model);
     m_internalModel = dynamic_cast<FileModel *>(model);
+    QSortFilterProxyModel::setSourceModel(model);
 }
 
 FileModel *SortFilterFileModel::fileSourceModel() const
-- 
GitLab

